### PR TITLE
Created test for operatorhub

### PR DESCRIPTION
**Pull request by mmajerni**

### CODE REVIEW

File: src/test/java/apicurito/tests/steps/CommonSteps.java [hunk 1]
```diff
@@ -148,12 +150,49 @@ public void createParameterOnPage(String param, String page, String isPlus, Data
...
         }
         CommonUtils.fillEntityEditorForm(table);
     }
+
+    @When("deploy apicurito into operatorhub and subscribe")
+    public void deployApicuritoIntoOperatorhubAndSubscribe() {
+        String disableOH = "https://gist.githubusercontent.com/mmajerni/1ea94f02a6b66dd86ebe63d8e8a9f15e/raw/29801d1e281f85193408bfb34d971edbddd67d63/OHdisable.yaml";
+        String operatorsource = "https://gist.githubusercontent.com/mmajerni/17e4c892f0685e0b9eb2b7051af0494f/raw/5b40b8b45ab9e24afd0c89be6fa0e6ab7b2fa10a/operatorsource.yaml";
+        String operatorGroup = "https://gist.githubusercontent.com/mmajerni/b170fcb959c046633f91dda693f3396e/raw/1b5f373abadc2decba00c07530a055311b9ad01f/OHoperatorgroup.yaml";
+        String subscription = "https://gist.githubusercontent.com/mmajerni/25523c2e5cd720d99730ec0b4cf0cb96/raw/818e9f3683a5e5de5e0582598d3523ca03c9b01b/OHsubscription.yaml";
```
Comment (asmigala) [inline src/test/java/apicurito/tests/steps/CommonSteps.java]: Wouldn't it be better to have these yaml files stored locally in `src/test/resources`?

File: src/test/java/apicurito/tests/steps/CommonSteps.java [hunk 2]
```diff
@@ -148,12 +150,49 @@ public void createParameterOnPage(String param, String page, String isPlus, Data
...
+    @Then("check that apicurito operator is deployed and in running state")
+    public void checkThatApicuritoOperatorIsDeployedAndInRunningState() {
+        List<Pod> pods = OpenShiftUtils.getInstance().getPods();
+        for (Pod pod : pods) {
+            if (pod.getMetadata().getName().contains("apicurito-operator")) {
+                assertThat(pod.getStatus().getPhase()).isEqualTo("Running");
+                return;
+            }
+        }
+        fail("Apicurito operator pod does not exists.");
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/steps/CommonSteps.java]: fixed

File: src/test/java/apicurito/tests/steps/CommonSteps.java [hunk 3]
```diff
@@ -148,12 +150,49 @@ public void createParameterOnPage(String param, String page, String isPlus, Data
...
+        try {
+            Thread.sleep(15000L);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+        ApicuritoTemplate.applyOnOCP("Operator group", "operatorhub", operatorGroup);
+        ApicuritoTemplate.applyOnOCP("Subscription", "operatorhub", subscription);
+        //wait for OCP to create subscription,csv and apicurito operator pod
+        try {
+            Thread.sleep(30000L);
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/steps/CommonSteps.java]: added dynamic wait

File: src/test/java/apicurito/tests/steps/CommonSteps.java [hunk 4]
```diff
@@ -148,12 +150,49 @@ public void createParameterOnPage(String param, String page, String isPlus, Data
...
         }
         CommonUtils.fillEntityEditorForm(table);
     }
+
+    @When("deploy apicurito into operatorhub and subscribe")
+    public void deployApicuritoIntoOperatorhubAndSubscribe() {
+        String disableOH = "src/test/resources/operatorhubFiles/disableOH.yaml";
+        String operatorsource = "src/test/resources/operatorhubFiles/operatorsource.yaml";
+        String operatorGroup = "src/test/resources/operatorhubFiles/operatorGroup.yaml";
+        String subscription = "src/test/resources/operatorhubFiles/subscription.yaml";
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/steps/CommonSteps.java]: added

File: src/test/java/apicurito/tests/steps/CommonSteps.java [hunk 5]
```diff
@@ -148,12 +150,49 @@ public void createParameterOnPage(String param, String page, String isPlus, Data
...
+
+        ApicuritoTemplate.applyOnOCP("Disable operators", "openshift-marketplace", disableOH);
+        ApicuritoTemplate.applyOnOCP("Operator source", "openshift-marketplace", operatorsource);
+        //wait for a new operator
+        try {
+            Thread.sleep(15000L);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/steps/CommonSteps.java]: added dynamic wait

File: src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java [hunk 6]
```diff
@@ -94,6 +94,13 @@ private static void deployOnOCP(String itemName, String item) {
         );
     }
 
+    public static void applyOnOCP(String itemName, String namespace, String item) {
+        log.info("Applying " + itemName + " from: " + item);
```
[the remark this answers is not in the document]
Comment (mmajerni) [reply src/test/java/apicurito/tests/configuration/templates/ApicuritoTemplate.java]: changed

**Pull request merged**